### PR TITLE
VSICurl and derived: replace most uses of CPLString with std::string

### PR DESCRIPTION
**Pull request by rouault**
and a few coverity fixes

### CODE REVIEW

File: ogr/utf8.h [hunk 1]
```diff
@@ -1422,9 +1422,11 @@ utf8_constexpr14_impl utf8_int32_t utf8lwrcodepoint(utf8_int32_t cp) {
     case 0x01ac:
       cp = 0x01ad;
       break;
+#if 0
```
Comment (elpaso) [inline ogr/utf8.h:1425]: leftover?

**Review by elpaso — APPROVED** [review]
I think that moving away from CPLString is a good move but can you explain me what is the long term strategy?

Is the CPL* a legacy of the past when the standard library wasn't mature (or even existing)?

Is it the goal to eventually drop CPL completely?

**Comment by rouault** [comment]
> Is it the goal to eventually drop CPL completely?

no, I don't think so as it provides useful utilities. But using plain std::string when we don't need CPLString makes sense.

File: ogr/utf8.h (hunk 1 above)
Comment (rouault) [reply ogr/utf8.h:1425]: I had left that as a way to trace back the change w.r.t the upstream project, but I just submitted the removal upstream in https://github.com/sheredom/utf8.h/pull/121. I'll delete that code once upstream has integrated the change

Comment (rouault) [reply ogr/utf8.h]: done

**Pull request merged**